swaybar/ipc.c

```diff
@@ -322,8 +335,11 @@ void ipc_bar_init(struct bar *bar, const char *bar_id) {
 	}
 	free(res);
 	json_object_put(outputs);
-
-	const char *subscribe_json = "[ \"workspace\", \"mode\" ]";
+	// Will need to be redone for display mode toggling compatibility
+	// Will require setting bar->config->hidden_state to "show" as default
+	const char *subscribe_json = strcmp(bar->config->display_mode, "hide") == 0 ?
+		"[ \"workspace\", \"mode\", \"modifier\" ]"
+		: "[ \"workspace\", \"mode\" ]";
```

Comment:
@JerziKaminsky I had considered doing it that way specifically for the reason that I commented - if the user wants to toggle the display mode between "hide" and "dock" the bar will need to be able to get modifier events even if it started as a "dock" bar. I'll go ahead and do that. Not entirely sure what governs which modifiers get sent when. In any case, subscribe and ignore will work for now.